openapi3/refs.go

```diff
@@ -290,6 +291,12 @@ func (value *SchemaRef) UnmarshalJSON(data []byte) error {
 // Validate returns an error if SchemaRef does not comply with the OpenAPI spec.
 func (value *SchemaRef) Validate(ctx context.Context) error {
 	if v := value.Value; v != nil {
+		if x := v.Example; x != nil {
+			if err := ValidateExampleValue(ctx, x, v); err != nil {
+				return fmt.Errorf("invalid schema example: %s", err)
+			}
+		}
+
```

Comment:
Actually this must go within `func (schema *Schema) Validate(ctx context.Context) error {...}` otherwise validating schemas without dereferencing a `$ref` won't go through this code.